addons/main/functions/fnc_getLauncherUnits.sqf

```diff
@@ -0,0 +1,73 @@
+#include "script_component.hpp"
+/*
+ * Author: ThomasAngel
+ * Gets all units from a group that have launchers.
+ * Things like AP launchers, flares etc will not count, -
+ * launcher must be capable of destroying at least some vehicle.
+ *
+ * Checking through the submunitions might improve mod compatibility,
+ * but only if the AI actually accounts for submunitions when evaluating -
+ * whether to engage or not. As of writing, this is unknown.
+ *
+ * Disposable launchers get mostly ignored, AI doesn't seem to even use them.
+ *
+ * Arguments:
+ * 0: Group to search in <GROUP>
+ * 1: Check through submunitions <BOOLEAN>
+ *
+ * Return Value:
+ * Array - units with launchers
+ *
+ * Example:
+ * [group bob] call lambs_main_fnc_getLauncherUnits;
+ *
+ * Public: Yes
+*/
+
+#define LIGHT_VEHICLE "128"
+#define AIR_VEHICLE "256"
+#define HEAVY_VEHICLE "512"
+
+params [
+    ["_group", grpNull, [grpNull]],
+    ["_checkSubmunition", false, [false]],
+    ["_offensiveVeh", true, [true]],
+    ["_offensiveAir", true, [true]],
+    ["_offensiveArmor", true, [true]]
+];
+
+private _suitableUnits = [];
+{
+    if ((secondaryWeapon _x) isEqualTo "") then {continue};
+    private _currentUnit = _x;
+
+    private _unitsMagazines = (magazines _currentUnit) + (secondaryWeaponMagazine _currentUnit);
+    {
+        private _mainAmmo = getText (configFile >> "cfgMagazines" >> _x >> "ammo");
+        private _mainAmmoFlags = getText (configFile >> "cfgAmmo" >> _mainAmmo >> "aiAmmoUsageFlags");
```

Comment:
those values should probably be cached and additionally aiAmmoUsageFlags is not text but a number

Reply:
aiAmmoUsageFlags is kind of both. For the purposes of this function, it is needed in text form.

``getText (configFile >> "CfgAmmo" >> "M_Vorona_HEAT" >> "aiAmmoUsageFlags") // Returns "128 + 512"``
``getNumber (configFile >> "CfgAmmo" >> "M_Vorona_HEAT" >> "aiAmmoUsageFlags") // Returns 640``

Reply:
aiAmmoUsageFlags are bitflags and should also be used that way. this is also the reason why you don't get RHS Disposable and possibly other launchers because they may write use 640 for example